includes/class-wc-session-handler.php

```diff
@@ -90,6 +90,11 @@ public function init() {
 		add_action( 'shutdown', array( $this, 'save_data' ), 20 );
 		add_action( 'wp_logout', array( $this, 'destroy_session' ) );
 
+		// Support for the User Switching plugin.
+		add_action( 'switch_to_user', array( $this, 'forget_session' ) );
```

Comment:
I do not think these actions should be added in WooCommerce since this is very specific to user switching. The plugin should be able to achieve the same with the following code:
```
add_action( 'switch_to_user', array( WC()->session, 'destroy_session' ) );
```
The specific plugin can also add their own function to clear just the cookie, the cookie name is simply `'wp_woocommerce_session_' . COOKIEHASH`

Reply:
I understand your concern about adding plugin-specific code to WooCommerce, but I'm in the same boat with User Switching 😁.

As I mentioned in the description, there's currently no public method that another plugin can call to forget a session without destroying it. Your code example above causes the session to be destroyed, not just forgotten.

I looked into manually clearing the session cookie, but again this is susceptible to breakage in the future if WC changes the name of the cookie, or changes the way sessions are handled.

A public method for forgetting the session and the cart is the safest contract between the two plugins.

Reply:
> A public method for forgetting the session and the cart is the safest contract between the two plugins.

I agree with this, so let's update the session handler class in this PR and introduce a new method that clears the session without destroying it as you have already. Then you can introduce the hooks in your plugin to call that method on your actions? That way you won't have to worry about cookie name changes.

Reply:
While I would prefer everything to live within WooCommerce, I can get behind that. I'll update this PR to remove the User Switching hooks.

Reply:
Done.